dali/imgcodec/util/convert_gpu_test.cc

```diff
@@ -298,6 +281,119 @@ TYPED_TEST(ConvertGPUTest, RGBToGray) {
   this->CheckConvert("HWC", DALI_GRAY, "HWC", DALI_RGB);
 }
 
+TYPED_TEST(ConvertGPUTest, Rotation90) {
+  this->SetInput({
+    {pixelA, pixelB, pixelC},
+    {pixelD, pixelE, pixelF},
+  });
+
+  this->SetReference({
+    {pixelD, pixelA},
+    {pixelE, pixelB},
+    {pixelF, pixelC},
+  });
+
+  this->CheckConvert("HWC", DALI_RGB, "HWC", DALI_RGB, {}, {90, false, false});
+}
+
+TYPED_TEST(ConvertGPUTest, Rotation90FlipX) {
+  this->SetInput({
+    {pixelA, pixelB, pixelC},
+    {pixelD, pixelE, pixelF},
+  });
+
+  this->SetReference({
+    {pixelA, pixelD},
+    {pixelB, pixelE},
+    {pixelC, pixelF},
+  });
+
+  this->CheckConvert("HWC", DALI_RGB, "HWC", DALI_RGB, {}, {90, true, false});
+}
+
+TYPED_TEST(ConvertGPUTest, Rotation180) {
+  this->SetInput({
+    {pixelA, pixelB, pixelC},
+    {pixelD, pixelE, pixelF},
+  });
+
+  this->SetReference({
+    {pixelF, pixelE, pixelD},
+    {pixelC, pixelB, pixelA},
+  });
+
+  this->CheckConvert("HWC", DALI_RGB, "HWC", DALI_RGB, {}, {180, false, false});
+}
+
+TYPED_TEST(ConvertGPUTest, Rotation270) {
+  this->SetInput({
+    {pixelA, pixelB, pixelC},
+    {pixelD, pixelE, pixelF},
+  });
+
+  this->SetReference({
+    {pixelC, pixelF},
+    {pixelB, pixelE},
+    {pixelA, pixelD},
+  });
+
+  this->CheckConvert("HWC", DALI_RGB, "HWC", DALI_RGB, {}, {270, false, false});
+}
+
+TYPED_TEST(ConvertGPUTest, FlipX) {
+  this->SetInput({
+    {pixelA, pixelB, pixelC},
+    {pixelD, pixelE, pixelF},
+  });
+
+  this->SetReference({
+    {pixelC, pixelB, pixelA},
+    {pixelF, pixelE, pixelD},
+  });
+
+  this->CheckConvert("HWC", DALI_RGB, "HWC", DALI_RGB, {}, {0, true, false});
+}
+
+TYPED_TEST(ConvertGPUTest, FlipY) {
+  this->SetInput({
+    {pixelA, pixelB, pixelC},
+    {pixelD, pixelE, pixelF},
+  });
+
+  this->SetReference({
+    {pixelD, pixelE, pixelF},
+    {pixelA, pixelB, pixelC},
+  });
+
+  this->CheckConvert("HWC", DALI_RGB, "HWC", DALI_RGB, {}, {0, false, true});
+}
+
+TYPED_TEST(ConvertGPUTest, TransposeAndRotate) {
```

Comment:
It's not immediately obvious why this tests uses literals. I'd suggest a more explicit name:
```suggestion
TYPED_TEST(ConvertGPUTest, TransposeAndRotate_PlanarToInterleaved) {
```
or
```suggestion
TYPED_TEST(ConvertGPUTest, TransposeAndRotate_CHw2HWC) {
```